### PR TITLE
Fix another batch of Sonar-identified issues

### PR DESCRIPTION
**Pull request by idodeclare**
- Sanitize several user-supplied strings.
- Fix some scope issues.
- Prohibit XML external access.

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java [hunk 1]
```diff
@@ -561,7 +554,6 @@ private void findFilelessChildren(SkeletonDirs skels, File directory) {
 
     /**
      * Counts segments arising from {@code File.separatorChar} or '\\'.
-     * @param path
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java:564]: why is this deleted ?

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+     * Sanitize {@code value} where it will be used in subsequent OpenGrok
+     * (non-logging) processing.
+     * @return {@code null} if null or else {@code value} with "pattern-breaking
+     * characters" (tabs, CR, LF, FF) replaced as underscores (one for one)
+     */
+    public static String userInput(String value) {
+        if (value == null) {
+            return null;
+        }
+        return value.replaceAll("[\\n\\r\\t\\f]", "_");
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java:45]: reuse the String literal ?

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+     * collisions, the colliding keys' values are combined.
+     */
+    public static Map<String, String[]> logging(Map<String, String[]> map) {
+        if (map == null) {
+            return null;
+        }
+
+        HashMap<String, String[]> safes = new HashMap<>();
+        for (Map.Entry<String, String[]> entry : map.entrySet()) {
+            String k = logging(entry.getKey());
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java:90]:
```suggestion
            String key = logging(entry.getKey());
```

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+package org.opengrok.indexer.web;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a container for sanitizing methods for avoiding classifications as
+ * taint bugs.
+ */
+public class LaunderUtil {
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java:33]: perhaps this class should be called Laundromat and have the launder as a method ? :-)

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java (hunk 2 above)
Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java:45]: also, userInput() and luceneQuery() can be refactored into using common underlying method ?

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java (hunk 4 above)
Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java:33]: also, add a simple test ?

File: opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java (hunk 1 above)
Comment (idodeclare) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java:564]: Restored with a comment

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java (hunk 2 above)
Comment (idodeclare) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java]: Yes done

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java (hunk 3 above)
Comment (idodeclare) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java]: OK done

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java (hunk 4 above)
Comment (idodeclare) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java]: You got it

**Comment by tarzanek** [comment]
I could argue about naming of Laundromat and its methods, but I guess I'll just keep smiling
feel free to merge from my side

**Comment by idodeclare** [comment]
Thank you for taking a look, Lubos 😁

**Comment by tarzanek** [comment]
merging

**Comment by idodeclare** [comment]
Thank you, Lubos
